controller/src/main/java/io/pravega/controller/fault/ControllerClusterListener.java

```diff
@@ -74,18 +88,6 @@ protected void startUp() throws InterruptedException, Exception {
             log.info("Registering host {} with controller cluster", host);
             cluster.registerHost(host);
 
-            if (eventProcessorsOpt.isPresent()) {
```

Comment:
what we are doing here is to make sure that we always start clusterlisterner without depending on other components. 
I am thinking in future i will generalize all sweepers using an interface so that its all streamlined, 

Currently all three sweepers each implement similar methods but are doing things differently. 
Using a queue streamlines that.. and ensures that such issues are not encountered in future because future changes to this class will implicitly take care of ensuring that sweepers wait on independent components.. respective sweepers schedule sweeps as and when they become ready.. thus little risk of any similar mistakes in future.